tests/plugins_t/test_elasticsearchmonitoring.py

```diff
@@ -0,0 +1,81 @@
+from tardis.plugins.elasticsearchmonitoring import ElasticsearchMonitoring
+from tardis.utilities.attributedict import AttributeDict
+from tardis.interfaces.siteadapter import ResourceStatus
+from tardis.resources.dronestates import CleanupState
+
+from datetime import datetime
+from unittest import TestCase
+from unittest.mock import Mock
+from unittest.mock import patch
+
+from ..utilities.utilities import run_async
+
+
+class TestElasticsearchMonitoring(TestCase):
+    @classmethod
+    def setUpClass(cls):
+        cls.mock_config_patcher = patch(
+            "tardis.plugins.elasticsearchmonitoring.Configuration"
+        )
+        cls.mock_elasticsearch_patcher = patch(
+            "tardis.plugins.elasticsearchmonitoring.Elasticsearch"
+        )
+        cls.mock_datetime_patcher = patch(
+            "tardis.plugins.elasticsearchmonitoring.datetime"
+        )
+        cls.mock_time_patcher = patch("tardis.plugins.elasticsearchmonitoring.time")
+        cls.mock_config = cls.mock_config_patcher.start()
+        cls.mock_elasticsearch = cls.mock_elasticsearch_patcher.start()
+        cls.mock_datetime = cls.mock_datetime_patcher.start()
+        cls.mock_time = cls.mock_time_patcher.start()
+
+    @classmethod
+    def tearDownClass(cls):
+        cls.mock_config_patcher.stop()
+        cls.mock_elasticsearch_patcher.stop()
+        cls.mock_datetime_patcher.stop()
+        cls.mock_time_patcher.stop()
+
+    @patch("tardis.plugins.elasticsearchmonitoring.logging", Mock())
+    def setUp(self):
+        self.config = self.mock_config.return_value
+        self.config.Plugins.ElasticsearchMonitoring._index = "index"
+        self.config.Plugins.ElasticsearchMonitoring._meta = "meta"
+
+        self.plugin = ElasticsearchMonitoring()
+
+    @patch("tardis.plugins.elasticsearchmonitoring.logging", Mock())
+    def test_notify(self):
+        test_param = AttributeDict(
+            site_name="test-site",
+            machine_type="test_machine_type",
+            created=datetime.now(),
+            updated=datetime.now(),
+            resource_status=ResourceStatus.Booting,
+            drone_uuid="test-drone",
+        )
+
+        test_param_ext = {
+            **test_param,
+            "state": str(CleanupState()),
+            "meta": self.plugin._meta,
+            "timestamp": int(self.mock_time.return_value * 1000),
+            "resource_status": str(test_param.resource_status),
+            "revision": 1,
```

Comment:
I do not understand where the "revision": 1 comes from. Maybe you can give me a hint. I see that `self.mock_elasticsearch.return_value.search` is called from https://github.com/MatterMiners/tardis/blob/e48d07a8fd0fcaecf251477a7da0269e209201e3/tardis/plugins/elasticsearchmonitoring.py#L69, but its return value is not set anywhere in the test.

Reply:
Sorry, I'm afraid I don't understand the question. `revision` is set in [this line](https://github.com/MatterMiners/tardis/blob/426da4ea01e88cd2097c1cd28341aa0db104f34e/tardis/plugins/elasticsearchmonitoring.py#L77).  Do I need to set the return value of `self._es.search(...)`?

Reply:
You are testing the `notifiy` call 
https://github.com/MatterMiners/tardis/blob/426da4ea01e88cd2097c1cd28341aa0db104f34e/tardis/plugins/elasticsearchmonitoring.py#L30-L53
this will call via `async_execute` and `execute` the elasticsearch function `search`. You are mocking this function, so it should return a revision, which you have to set in the tests itself via `self.mock_elasticsearch.return_value.search.return_value`. Afterwards the elasticsearch function `create` is called with the revision in its document. You can then check in https://github.com/MatterMiners/tardis/blob/426da4ea01e88cd2097c1cd28341aa0db104f34e/tests/plugins_t/test_elasticsearchmonitoring.py#L74-L78 that the call contains the right parameters.

Reply:
Now I understand, thanks a lot! I was only lucky that this test passed before. I fixed it, and I changed `revision` from `1` to `2`, to make it more exciting.